### PR TITLE
 Don't merge files if the download is incomplete & code cleanup #43 

### PR DESCRIPTION
**Pull request by arvyshka**
- Remove global variables and pass as dependency 
- Fixed: https://github.com/arvyshka/blazer/issues/30
- Fixed: https://github.com/arvyshka/blazer/issues/42

### CODE REVIEW

File: internals/network/concurrent_downloader.go [hunk 1]
```diff
@@ -20,44 +20,45 @@ func acceptedStatusCodes(code int) bool {
...
 		i := i
 		segment := segment
+
+		// Downoading individual segments in go routines
 		go func() {
 			defer wg.Done()
-			err = DownloadSegment(request, i, segment)
+			err = DownloadSegment(request, i, segment, sessionID)
 			if err != nil {
+				isDownloadComplete = false
```
Comment (setraj) [inline internals/network/concurrent_downloader.go:51]: Here, why are we sharing a resource(isDownloadComplete) among multiple goroutines?
This is a clear invitation for the RACE condition.

see this:  https://stackoverflow.com/questions/39207608/how-does-golang-share-variables-between-goroutines

File: pkg/data/utils.go [hunk 2]
```diff
@@ -8,10 +8,10 @@ import (
 	"io"
 	"os"
 	"strings"
-
-	"github.com/arvyshka/blazer/internals"
 )
 
+const MemUnit = 1024
```
Comment (setraj) [inline pkg/data/utils.go:13]: for consistency across the project, we can follow all constants as CAPS policy.

File: cmd/blazer.go [hunk 3]
```diff
@@ -20,33 +22,35 @@ func Start() {
 		fmt.Println("Error parsing flags: ", err)
```
Comment (setraj) [inline cmd/blazer.go:22]: We should try to make all the free-flowing err/info msgs as constant strings.

**Review by setraj — CHANGES_REQUESTED** [review]

File: cmd/blazer.go (hunk 3 above)
Comment (arvyshka) [reply cmd/blazer.go:22]: sure, will update it

File: internals/network/concurrent_downloader.go (hunk 1 above)
Comment (setraj) [reply internals/network/concurrent_downloader.go:51]: Right, let's figure out a better way to achieve the same.

File: pkg/data/utils.go (hunk 2 above)
Comment (arvyshka) [reply pkg/data/utils.go:13]: @setraj : our linters take care of it, to start it you need to run `make lint`. We need to make it part of our CI/CD flow. I tried to integrate github's CI/CD but it is not working as expected.. have you used github's CI/CD if not what do you suggest ?

File: cmd/blazer.go (hunk 3 above)
Comment (arvyshka) [reply cmd/blazer.go:22]: please create an issue for it. we need to change in many places

**Comment by arvyshka** [comment]
@setraj : you finished reviewing this PR ? can i merge it ??

**Review by setraj — APPROVED** [review]

**Pull request merged**